Std/Data/HashMap/Lemmas.lean

```diff
@@ -4,12 +4,21 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
 -/
 import Std.Data.HashMap.Basic
+import Std.Data.Array.Lemmas
 import Std.Util.ProofWanted
 
 namespace Std.HashMap
 
-@[simp] proof_wanted empty_find? [BEq α] [Hashable α] {a : α} :
-    (∅ : HashMap α β).find? a = none
+namespace Imp
+
+@[simp] theorem empty_buckets : (empty : Imp α β).buckets = ⟨mkArray 8 AssocList.nil, by simp⟩ := rfl
```

Comment:
Do we really want to be hard-coding 8 here? It seems like either this should be a constant `Imp.emptyCapacity` or we should avoid stating it. If we want to avoid stating this lemma entirely, the `simp` invocation in `empty_find?` could be fixed be adding `[Imp.empty', Imp.empty, Imp.Buckets.mk]`.

Reply:
That's a good catch.  I think the approach of adding additional defs to unfold is good.

Alternatively, `empty_buckets` should work over any user-defined `capacity` value passed into `Imp.empty`.